prior_noise:
    R_IB: [0.03, 0.03, 0.03]
    I_p_IB: [0.01, 0.01, 0.01]
    I_v_IB: [0.01, 0.01, 0.01]
    # Acc bias repeatability from data sheet (BMI088 20mg -> 3-sigma = 0.2 m/s**2).
    b_a: [0.07, 0.07, 0.07]
    # Gyro bias up to sensitivity, because it is calibrated at startup. (BMI088 16.384 LSB/°/s @ 2000 °/s -> 3-sigma = 0.0011°/s)
    b_g: [3.6e-4, 3.6e-4, 3.6e-4]

noise:
    radar:
        doppler: 0.05 # sigma
        track: [0.1, 0.1, 0.1] # BearingRange3D TODO(rikba): determine order, probably azimuth, elevation, range 
        loss: 5 # 0: quadratic, 1: fair, 2: huber, 3: cauchy, 4: geman-mcclure, 5: welsch, 6: tukey
    baro:
        height: 0.2 # RMS/sqrt(dt) (BMP390 0.11m sigma, dt=0.02)
        loss: 1 # 0: quadratic, 1: fair, 2: huber, 3: cauchy, 4: geman-mcclure, 5: welsch, 6: tukey
radar:
    track_age: 10 # track for n frames, i.e., 0 = only current frame, 1 = current and previous frame, etc.

baro:
    active: True

imu:
    bias_acc_sigma: 1.0e-4
    bias_omega_sigma: 1.0e-5
    bias_acc_int_sigma: 1.0e-6
    bias_omega_int_sigma: 1.0e-6

    acc_sigma: 1.0e-1
    gyro_sigma: 1.0e-2
    integration_sigma: 1.0e-6

    initial_bias_acc: [-0.15, -0.21, 0.15]
    initial_bias_gyro: [0.0, 0.0, 0.0]

isam2:
    relinearize_threshold: 0.1
    relinearize_skip: 10
    find_unused_factor_slots: False
    cache_linearized_factors: True
    enable_partial_relinarization_check: False
    smoother_lag: 5.0 # s WARNING: This time needs to be greater than any imaginable sensor arrival delay
    optimizer: 0 # 0: Gauss-Newton, 1: Dogleg
    gn:
        wildfire_threshold: 0.001
